### PR TITLE
fix: on comment thread show mention suggestions when user discards the invitation flow 

### LINKED ISSUE #8193: [Bug] Tagging Appsmith support on comment mode
## Description
Press **Esc** when the invite pop-up shows up when Appsmith support is tagged and then continuing to comment again. Observe that the invite option does not appear again

### Steps to reproduce the behaviour:
1. Go to **comment mode** and tag Appsmith support. Observe the invite pop-up
2. Now press **Esc**/click outside the box to close the box 
3. Tag support again in a comment and observe no pop-up appear

### Important Details
- Version: Cloud
- OS: Win10
- Browser: Chrome
- Environment: Prod

[LOOM DEMO](https://www.loom.com/share/5d0eac64c7a44d18a99a5c6861cf007e)

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Updating the comment value object to discard the mention tag for current invitation candidate.

Fixes #8193

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- automated test added

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/select-mentions-on-discard-share-modal 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.75 **(-0.01)** | 36.62 **(0)** | 33.45 **(-0.02)** | 55.28 **(-0.02)**
 :red_circle: | app/client/src/comments/inlineComments/AddCommentInput.tsx | 60.56 **(-4.44)** | 19.82 **(-0.18)** | 56 **(-4.87)** | 62.18 **(-4.97)**
 :red_circle: | app/client/src/comments/inlineComments/InlineCommentPin.tsx | 86.96 **(-0.4)** | 75 **(-0.76)** | 94.44 **(0)** | 89.29 **(0.54)**
 :red_circle: | app/client/src/pages/organization/OrgInviteUsersForm.tsx | 38.71 **(-1.48)** | 24 **(0)** | 0 **(0)** | 39.32 **(-2.43)**</details>

### CODE REVIEW

File: app/client/src/comments/inlineComments/AddCommentInput.tsx [hunk 1]
```diff
@@ -215,9 +221,43 @@ function AddCommentInput({
   );
 
   const setEditorState = useCallback((updatedEditorState: EditorState) => {
-    setEditorStateInState(updatedEditorState);
+    const currentURL = new URL(window.location.href);
+    const searchParams = currentURL.searchParams;
+    const isInviting = searchParams.get("inviting");
+    let modifiedState = updatedEditorState;
+    if (isInviting) {
```
Comment (rishabhsaxena) [inline app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: nit: wondering if we could do this conditionally, if the invitation is successful we remove the tag else we keep it? Or I guess if we could auto-focus the input.
Also a QQ: would things always happen in this flow, just to check if there can be a race condition here?

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/comments/inlineComments/AddCommentInput.tsx (hunk 1 above)
Comment (pranavkanade) [reply app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: Actually, figuring out `invitation success` is not needed as in the end user has to close the modal manually. The race condition shouldn't be a problem, as the control will from modal to canvas. I did think about auto-focus, but couldn't find a proper place for it.

File: app/client/src/pages/organization/OrgInviteUsersForm.tsx [hunk 2]
```diff
@@ -203,6 +204,28 @@ const validate = (values: any) => {
...
+export const finalizeInviteFromComment = () => {
+  const currentURL = new URL(window.location.href);
+  const searchParams = currentURL.searchParams;
+  searchParams.delete("inviting");
+  history.replace({
+    ...window.location,
+    pathname: currentURL.pathname,
+    search: searchParams.toString(),
+  });
+};
```
[the remark this answers is not in the document]
Comment (pranavkanade) [reply app/client/src/pages/organization/OrgInviteUsersForm.tsx:227]: As there won't be a race condition, we don't need this.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js [hunk 3]
```diff
@@ -10,24 +10,6 @@ let appName;
...
-        cy.wait("@createOrg").then((interception) => {
-          const newOrganizationName = interception.response.body.data.name;
-          cy.renameOrg(newOrganizationName, orgName);
-        });
-        cy.CreateAppForOrg(orgName, appName);
-        cy.addDsl(dsl);
-      });
-    });
-  });
-
```
[the remark this answers is not in the document]
Comment (pranavkanade) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:30]: restoring

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js [hunk 4]
```diff
@@ -71,6 +53,20 @@ describe("Comments", function() {
...
+      .type("@appsmith", {
+        delay: 100,
+      })
+      .type("{enter}");
+    cy.contains("support@appsmith.com").should("be.visible");
+    cy.get(homePage.closeBtn).click();
+    cy.wait(1000);
+    cy.get(commentsLocators.mentionsInput)
+      .focus()
+      .type("{movetoend}");
```
[the remark this answers is not in the document]
Comment (pranavkanade) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Comments/AddComments_spec.js:67]: https://github.com/appsmithorg/appsmith/pull/9062#discussion_r746271986

File: app/client/src/comments/inlineComments/AddCommentInput.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: Got it, so currently we have to click on the suggestion item again to tag in either of cases if the user was/wasn't invited? was wondering if we could do it conditionally based on the invite success

Comment (pranavkanade) [reply app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: tried doing something like following, but it places the cursor at the beginning of the input field - 
```
  useEffect(() => {
    if (window.location.href) {
      const searchParams = new URL(window.location.href).searchParams;
      const isInviting = searchParams.get("inviting");
      if (!isInviting) {
        focusInput();
      }
    }
  }, [location]);
```

Any suggestions on how to move the control to the end of the value?

Comment (rishabhsaxena) [reply app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: https://github.com/brijeshb42/medium-draft/issues/71
Haven't tried this, but it looks like we could use `draft-js` apis

Comment (rishabhsaxena) [reply app/client/src/comments/inlineComments/AddCommentInput.tsx:228]: Also, still feel having to click again after inviting might be a bit confusing

**Comment by Rishabh-Rathod** [comment]
@pranavkanade  For some reason, I wasn’t able to reproduce this issue on `app.appsmith.com` using steps to reproduce mentioned  #8193

**Comment by pranavkanade** [comment]
@Rishabh-Rathod the `Esc` press will takes user to the `EDIT` mode. That repro step is incorrect. Although, notice how even after you close the modal without inviting, you didn't see the mentions suggestion again? that's the bug.

**Comment by pranavkanade** [comment]
@rishabhsaxena I've been through draftjs docs to solve this issue. The solution I've come up with seems way too flaky. Rather than manually editing the editor state, I'd like to find an alternative but direct way to `UNDO` the mention tagging. Abandoning the current approach and closing this PR for now. Will raise a new one, as soon as I find a better way.

**Pull request closed without merging**